### PR TITLE
11 audio panic on windows

### LINKED ISSUE #11: Audio Panic on Windows
Hi team,

trying to build this on a Windows machine and getting an error with the audio config that's created. It seems not every device would support the 44100 sample_rate and cpal expects that you match the output of the device that's chosen. 

It seems some resampling would need to take place before the step of outputting to the cpal audio device.

### PR DESCRIPTION
**Pull request by Amjad50**
Fixes #11

### CODE REVIEW

**Comment by Amjad50** [comment]
can you test this @dcvz? Thanks

File: src/audio.rs [hunk 1]
```diff
@@ -57,7 +127,60 @@ impl AudioPlayer {
...
+            wfs
+        }
+
+        /// Helper to merge channels into a single vector
+        fn write_frames(waves: Vec<Vec<f32>>, outbuffer: &mut Vec<f32>, channels: usize) {
+            let nbr = waves[0].len();
+            for frame in 0..nbr {
+                for chan in 0..channels {
+                    let value = waves[chan][frame];
+                    outbuffer.push(value);
```
Comment (dcvz) [inline src/audio.rs]:
```suggestion
                for wave in waves.iter().take(channels) {
                    outbuffer.push(wave[frame]);
```

File: src/audio.rs [hunk 2]
```diff
@@ -57,7 +127,60 @@ impl AudioPlayer {
     }
 
     pub fn queue(&mut self, data: &[f32]) {
-        self.buffer_producer.push_slice(data);
+        // helper method to split channels into separate vectors
+        fn read_frames(inbuffer: &Vec<f32>, n_frames: usize, channels: usize) -> Vec<Vec<f32>> {
```
Comment (dcvz) [inline src/audio.rs]:
```suggestion
        fn read_frames(inbuffer:  &[f32], n_frames: usize, channels: usize) -> Vec<Vec<f32>> {
```

**Review by dcvz — COMMENTED** [review]
perhaps some small suggestions?

**Pull request merged**